src/coreclr/gc/gc.cpp

```diff
@@ -49307,3 +49377,294 @@ void PopulateDacVars(GcDacVars *gcDacVars)
 #endif // MULTIPLE_HEAPS
     gcDacVars->generation_field_offsets = reinterpret_cast<int**>(&generation_field_offsets);
 }
+
+void GCHeap::RefreshMemoryLimit()
```

Comment:
we should really consider returning a status code here - what if the current committed is already higher than the new limit? should we try to do an aggressive GC? and if that still ends up being higher, what status should we return?